server/container_stop.go

```diff
@@ -38,6 +38,12 @@ func (s *Server) StopContainer(ctx context.Context, req *types.StopContainerRequ
 		return err
 	}
 
+	if s.config.EventedPLEG {
+		err := s.generateCRIEvent(ctx, c, types.ContainerEventType_CONTAINER_DELETED_EVENT)
+		if err != nil {
+			log.Errorf(ctx, "Unable to generate event %s for container %s due to err %s", types.ContainerEventType_CONTAINER_DELETED_EVENT, c.ID(), err)
```

Comment:
TODO item: should we emit a metric for events that were failed to generate?

Reply:
Yes, that's a great idea! Not per container but just having a counter collecting failures.